### PR TITLE
Update .travis.yml

### PR DESCRIPTION
**Pull request by John-Colvin**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1 +1,23 @@
+sudo: false
+
+os:
+ - linux
+ - osx
```
Comment (John-Colvin) [inline .travis.yml]: If you want the os x testers to actually run, you have to email travis-ci and ask them to enable them manually.

File: .travis.yml [hunk 2]
```diff
@@ -1 +1,23 @@
...
+ - dmd-2.068.2
+ - dmd-2.067.1
+ - dmd-2.066.1
+ - ldc-0.16.0-beta1
+ - ldc-0.15.2-beta2
+ - ldc-0.15.1
+ - gdc
+
+matrix:
+  allow_failures:
```
Comment (John-Colvin) [inline .travis.yml]: sadly there are 2 configurations that are currently broken on travis-ci

**Comment by John-Colvin** [comment]
I notice you don't actually have Tavis-CI enabled for this repository, so obviously you'd want to do that first.

**Comment by John-Colvin** [comment]
Hopefully you can see this page: https://travis-ci.org/John-Colvin/dlib

That's a lot of red :x:s, but it's mostly very simple little fixes that are needed.

**Comment by John-Colvin** [comment]
note that when you enable it it won't start testing anything until you make a new commit.

**Comment by gecko0307** [comment]
I see. BTW, does it treat warnings as errors?

**Comment by John-Colvin** [comment]
I think dub passes "-w" by default, I.e. warnings-as-errors.
you can use
```
"buildRequirements": "allowWarnings"
```
but I would strongly recommend fixing the warnings instead, as it will really help other people who like to build things all-at-once.

**Comment by John-Colvin** [comment]
FYI: by default travis seems to use gdc-4.9.2, which seems to segfault on your code. you could do
```
 - gdc-4.9.2
 - gdc-5.2.0
```
to check both of them. You would also have to explicitly list them in the allowed_failures as well.

The only reason I put the ldc-0.15.2-beta2 test is in there really is because it's currently the default in homebrew, there will never be an ldc-0.15.2 release, it's gonna go straight to 0.16.0

**Pull request merged**